Uplift/Controllers/FilterViewController.swift

```diff
@@ -333,47 +354,112 @@ class FilterViewController: UIViewController, RangeSeekSliderDelegate {
         }
 
+        classTypeDropdownFooter.snp.remakeConstraints { make in
+            make.top.equalTo(classTypeDropdown.snp.bottom)
+            make.left.right.equalToSuperview()
+
+            switch classTypeDropdownData.dropStatus! {
+            case .up:
+                make.height.equalTo(0)
+            case .half, .down:
+                make.height.equalTo(32)
+            }
+        }
+        
         classTypeInstructorDivider.snp.remakeConstraints { make in
             make.width.centerX.equalToSuperview()
-            make.top.equalTo(classTypeDropdown.snp.bottom)
-            make.bottom.equalTo(classTypeDropdown.snp.bottom).offset(1)
+            make.top.equalTo(classTypeDropdownFooter.snp.bottom)
+            make.bottom.equalTo(classTypeDropdownFooter.snp.bottom).offset(1)
         }
 
         // INSTRUCTOR SECTION
-        instructorDropdown.snp.remakeConstraints { make in
+        instructorDropdownHeader.snp.remakeConstraints { make in
             make.top.equalTo(classTypeInstructorDivider.snp.bottom)
             make.left.right.equalToSuperview()
+            make.height.equalTo(55)
+        }
+        
+        instructorDropdown.snp.remakeConstraints { make in
+            make.top.equalTo(instructorDropdownHeader.snp.bottom)
+            make.left.right.equalToSuperview()
 
             switch instructorDropdownData.dropStatus! {
             case .up:
-                make.height.equalTo(55)
+                make.height.equalTo(0)
             case .half:
-                make.height.equalTo(55 + 32 + 3*32)
+                make.height.equalTo(3*32)
             case .down:
-                make.height.equalTo(55 + 32 + instructorDropdown.numberOfRows(inSection: 0)*32)
+                make.height.equalTo(instructorDropdown.numberOfRows(inSection: 0)*32)
+            }
+        }
+        
+        instructorDropdownFooter.snp.remakeConstraints { make in
+            make.top.equalTo(instructorDropdown.snp.bottom)
+            make.left.right.equalToSuperview()
+
+            switch instructorDropdownData.dropStatus! {
+            case .up:
+                make.height.equalTo(0)
+            case .half, .down:
+                make.height.equalTo(32)
             }
         }
 
         instructorDivider.snp.remakeConstraints { make in
             make.width.centerX.equalToSuperview()
-            make.top.equalTo(instructorDropdown.snp.bottom)
+            make.top.equalTo(instructorDropdownFooter.snp.bottom)
             make.height.equalTo(1)
             make.bottom.equalToSuperview()
         }
+        
+        updateTableFooterViews()
+    }
+    
+    func setupDropdownHeaderViews() {
+        classTypeDropdownHeader.titleLabel.text = ClientStrings.Filter.selectClassTypeSection
+        instructorDropdownHeader.titleLabel.text = ClientStrings.Filter.selectInstructorSection
+        
+        let classGesture = UITapGestureRecognizer(target: self, action: #selector(self.dropClasses(sender:) ))
```

Comment:
not too sure if im understanding what this does but is `showClassesDropdown` a better name?

Reply:
I put toggleClasses because it's either show or hide